src/methods.jl

```diff
@@ -10,318 +19,378 @@ derivatives.
-    _, dfdx = fdm(m, f, x, Val(true); kwargs...)
-    return dfdx
+function _make_adaptive_bound_estimator(
+    constructor::Function,
+    q::Int,
+    adapt::Int,
+    condition::Int;
+    kw_args...
+)
+    if adapt >= 1
+        estimate_derivative =
+            constructor(q + 1, q, adapt=adapt - 1, condition=condition; kw_args...)
+        return (f, x) -> maximum(abs.(estimate_derivative(f, x)))
+    else
+        return _make_default_bound_estimator(condition=condition)
+    end
 end
 
+_forward_grid(p::Int) = 0:(p - 1)
 
-## Deprecations
+_backward_grid(p::Int) = (1 - p):0
 
-# Used for keyword argument name deprecations
-function _dep_kwarg(kwargs)
-    for (old, new) in [(:ε, :eps), (:M, :bound)]
-        haskey(kwargs, old) || continue
-        val = kwargs[old]
-        error(
-            "keyword argument `", old, "` should now be passed as `", new, "` upon ",
-            "application of the method. For example:\n    ",
-            "central_fdm(5, 1)(f, x; $new=$val)\n",
-            "not\n    ",
-            "central_fdm(5, 1; $old=$val)(f, x)"
-        )
+function _central_grid(p::Int)
+    if isodd(p)
+        return div(1 - p, 2):div(p - 1, 2)
+    else
+        return vcat(div(-p, 2):-1, 1:div(p, 2))
     end
 end
 
-function fdm(
-    grid::AbstractVector{<:Real},
-    q::Int,
-    ::Union{Val{true}, Val{false}}=Val(false);
-    kwargs...,
-)
-    error("to use a custom grid, use `Nonstandard(grid, q)` and pass the result to `fdm`")
-end
+_exponentiate_grid(grid::Vector, base::Int=3) = sign.(grid) .* base .^ abs.(grid) ./ base
 
-for fdmf in (:central_fdm, :backward_fdm, :forward_fdm)
-    @eval function $fdmf(p::Int, q::Int, ::Union{Val{true}, Val{false}}; kwargs...)
-        error(
-            "the `Val` argument should now be passed directly to `fdm` after ",
-            "constructing the method, not to the method constructor itself"
-        )
-    end
+"""
+    extrapolate_fdm(
+        m::FiniteDifferenceMethod,
+        f::Function,
+        x::T;
+        factor::Int=1000,
+        kw_args...
+    ) where T<:AbstractFloat
+
+Use Richardson extrapolation to refine a finite difference method. This method uses
+[`estimate_step`](@ref) to determine an appropriate initialisation for
+[`Richardson.extrapolate`](@ref).
+
+Takes further in keyword arguments for [`Richardson.extrapolate`](@ref).
+
+# Arguments
+- `m::FiniteDifferenceMethod`: Finite difference method to estimate the step size for.
+- `f::Function`: Function to evaluate the derivative of.
+- `x::T`: Point to estimate the derivative at.
+
+# Factor
+- `factor::Int=1000`: Factor to amplify the estimated step size by.
+
+# Returns
+- Estimate of the derivative.
+"""
+function extrapolate_fdm(
+    m::FiniteDifferenceMethod,
+    f::Function,
+    x::T;
+    factor::Int=1000,
+    kw_args...
+) where T<:AbstractFloat
+    h_conservative = estimate_step(m, f, x)[1] * factor
+    return extrapolate(h -> m(f, x, h), h_conservative; kw_args...)
```

Comment:
Could be beneficial to pass `power=2` to `extrapolate` if `m` is a symmetric method, so that only powers of h^2 are present in the Taylor series for the error.  Is this information available in `m`?

Reply:
Ah, yes, nice! It's not available explicitly, but we should be able to determine that. Let me just add that.